cesium/internal/domain/writer.go

```diff
@@ -162,17 +186,27 @@ func (w *Writer) Commit(ctx context.Context, end telem.TimeStamp) error {
 		fileKey:   w.fileKey,
 	}
 	f := lo.Ternary(w.prevCommit.IsZero(), w.idx.insert, w.idx.update)
+	err := f(ctx, ptr, persist)
+	if err != nil {
+		return span.Error(err)
+	}
+
 	w.prevCommit = end
-	return span.EndWith(f(ctx, ptr))
+	return nil
 }
 
 // Close closes the writer, releasing any resources it may have been holding. Any
-// uncommitted data will be discarded. Close is not idempotent, and is also not
-// safe to call concurrently with any other writer methods.
-func (w *Writer) Close() error {
+// uncommitted data will be discarded. Any committed, but unpersisted data will be persisted.
+// Close is idempotent, and is also not safe to call concurrently with any other writer methods.
+func (w *Writer) Close(ctx context.Context) error {
 	if w.closed {
 		return nil
 	}
+
+	if !w.AutoPersistTime.IsZero() {
+		w.idx.persist(ctx)
```

Comment:
I'm a bit sussed out about this. It doesn't need a mutex lock on the index?

Reply:
You're right it does